src/backend/InvenTree/plugin/builtin/barcodes/inventree_barcode.py

```diff
@@ -8,29 +8,45 @@
 """
 
 import json
+import re
+from typing import cast
 
 from django.utils.translation import gettext_lazy as _
 
+import plugin.base.barcodes.helper
 from InvenTree.helpers import hash_barcode
-from InvenTree.helpers_model import getModelsWithMixin
 from InvenTree.models import InvenTreeBarcodeMixin
 from plugin import InvenTreePlugin
-from plugin.mixins import BarcodeMixin
+from plugin.mixins import BarcodeMixin, SettingsMixin
 
 
-class InvenTreeInternalBarcodePlugin(BarcodeMixin, InvenTreePlugin):
+class InvenTreeInternalBarcodePlugin(SettingsMixin, BarcodeMixin, InvenTreePlugin):
     """Builtin BarcodePlugin for matching and generating internal barcodes."""
 
     NAME = 'InvenTreeBarcode'
     TITLE = _('InvenTree Barcodes')
     DESCRIPTION = _('Provides native support for barcodes')
-    VERSION = '2.0.0'
+    VERSION = '2.1.0'
     AUTHOR = _('InvenTree contributors')
 
-    @staticmethod
-    def get_supported_barcode_models():
-        """Returns a list of database models which support barcode functionality."""
-        return getModelsWithMixin(InvenTreeBarcodeMixin)
+    SETTINGS = {
+        'INTERNAL_BARCODE_FORMAT': {
+            'name': _('Internal Barcode Format'),
+            'description': _('Select an internal barcode format'),
+            'choices': [
+                ('json', _('JSON barcodes (require more space)')),
+                ('short', _('Short barcodes (made for optimized space)')),
+            ],
+            'default': 'json',
+        },
+        'SHORT_BARCODE_PREFIX': {
+            'name': _('Short Barcode Prefix'),
+            'description': _(
+                'Customize the prefix used for short barcodes, may be useful for environments with multiple InvenTree instances'
+            ),
+            'default': 'INV-',
```

Comment:
Good idea, this should work out of the box, I added some tests now to ensure this behavior.

Reply:
Sorry, this would be supported from my side, but isn't due to #7690, because maybe some codes then collide with the format of some suppliers. I removed it for now from the tests, but this can be added if that issue is solved.

Reply:
Should we allow setting it to empty then?

Reply:
I think for now not, as this doesn't work correctly. That's why I added the `MinLengthValidator(1)` to ensure the prefix is always at least one character long.

Reply:
Hm, that's really strange, I thought that would be an issue because the tests keeps failing, but only the sqlite one with the supplier barcode plugins. So I removed the empty prefix tests, but after a closer look, it is still failing. I need to investigate that a bit further tomorrow.

Reply:
@matmair Ok, I found the issue. This has nothing to do with supplier plugins. Was just thinking that because these tests were failing. The main issue was the `@lru_cache` decorator which cached an empty response. And somehow only the sqlite + coverage tests returned for the first time an empty list of barcode mixin models, which caused all further tests relying on that to fail.

I have now built my own cache decorator which does not cache empty results and re-added the ability to use an empty prefix. The tests are passing now.